types/tests/react.test.ts

```diff
@@ -11,9 +11,25 @@ const clientSession = {
   expires: "1234",
 }
 
-// $ExpectType [Session | null, boolean]
+/**
+ * $ExpectType
+ * | { data: Session; status: "authenticated"; loading: false; }
+ * | { data: null; status: "unauthenticated" | "loading"; loading: boolean; }
+ * | { //// data: Session; status: "authenticated"; loading: false; }
+ * | { data: null; status: "loading"; loading: true; }
+ */
 client.useSession()
 
+// $ExpectType { data: Session; status: "authenticated"; loading: false; } | { data: null; status: "loading"; loading: true; }
+const session = client.useSession({ required: true })
+if (session.status === "loading") {
+  // $ExpectType null
+  session.data
+} else {
+  // $ExpectType Session
+  session.data
+}
```

Comment:
Should we have a third case here?
```ts
const session = client.useSession({ required: false })
// $ExpectType Session
session.data

Reply:
In the case of `required: true`, there is only two valid status values. `loading` and `authenticated`.